vms/platformvm/camino_service.go

```diff
@@ -63,6 +85,135 @@ type GetConfigurationReply struct {
+		logging.UserStrings("addresses", args.Addresses),
+	)
+
+	// Parse to address
+	addrs, err := avax.ParseServiceAddresses(service.addrManager, args.Addresses)
+	if err != nil {
+		return err
+	}
+
+	utxos, err := avax.GetAllUTXOs(service.vm.state, addrs)
+	if err != nil {
+		return fmt.Errorf("couldn't get UTXO set of %v: %w", args.Addresses, err)
+	}
+
+	unlockedOutputs := map[ids.ID]uint64{}
+	bondedOutputs := map[ids.ID]uint64{}
+	depositedOutputs := map[ids.ID]uint64{}
+	depositedBondedOutputs := map[ids.ID]uint64{}
+
+utxoFor:
+	for _, utxo := range utxos {
+		assetID := utxo.AssetID()
+		switch out := utxo.Out.(type) {
+		case *secp256k1fx.TransferOutput:
+			newBalance, err := math.Add64(unlockedOutputs[assetID], out.Amount())
+			if err != nil {
+				unlockedOutputs[assetID] = stdmath.MaxUint64
+			} else {
+				unlockedOutputs[assetID] = newBalance
+			}
+		case *locked.Out:
+			switch out.LockState() {
+			case locked.StateBonded:
+				newBalance, err := math.Add64(bondedOutputs[assetID], out.Amount())
+				if err != nil {
+					bondedOutputs[assetID] = stdmath.MaxUint64
+				} else {
+					bondedOutputs[assetID] = newBalance
+				}
+			case locked.StateDeposited:
+				newBalance, err := math.Add64(depositedOutputs[assetID], out.Amount())
+				if err != nil {
+					depositedOutputs[assetID] = stdmath.MaxUint64
+				} else {
+					depositedOutputs[assetID] = newBalance
+				}
+			case locked.StateDepositedBonded:
+				newBalance, err := math.Add64(depositedBondedOutputs[assetID], out.Amount())
+				if err != nil {
+					depositedBondedOutputs[assetID] = stdmath.MaxUint64
+				} else {
+					depositedBondedOutputs[assetID] = newBalance
+				}
+			default:
+				service.vm.ctx.Log.Warn("Unexpected utxo lock state")
+				continue utxoFor
+			}
+		default:
+			service.vm.ctx.Log.Warn("unexpected output type in UTXO",
+				zap.String("type", fmt.Sprintf("%T", out)),
+			)
+			continue utxoFor
+		}
+
+		response.UTXOIDs = append(response.UTXOIDs, &utxo.UTXOID)
+	}
+
+	balances := map[ids.ID]uint64{}
+	for assetID, amount := range unlockedOutputs {
+		newBalance, err := math.Add64(balances[assetID], amount)
+		if err != nil {
+			balances[assetID] = stdmath.MaxUint64
+		} else {
+			balances[assetID] = newBalance
+		}
+	}
+	for assetID, amount := range bondedOutputs {
+		newBalance, err := math.Add64(balances[assetID], amount)
+		if err != nil {
+			balances[assetID] = stdmath.MaxUint64
+		} else {
+			balances[assetID] = newBalance
+		}
+	}
+	for assetID, amount := range depositedOutputs {
+		newBalance, err := math.Add64(balances[assetID], amount)
+		if err != nil {
+			balances[assetID] = stdmath.MaxUint64
+		} else {
+			balances[assetID] = newBalance
+		}
+	}
+	for assetID, amount := range depositedBondedOutputs {
+		newBalance, err := math.Add64(balances[assetID], amount)
+		if err != nil {
+			balances[assetID] = stdmath.MaxUint64
+		} else {
+			balances[assetID] = newBalance
+		}
+	}
+
+	response.Balances = newJSONBalanceMap(balances)
```

Comment:
I think it would be better to make balances maps json.Uint64 and cast when its needed, then to allocate another map and copy values with newJSONBalanceMap

Reply:
Not sure what you mean exactly. Do you mean changing
balances := map[ids.ID]uint64{}
to 
balances := map[ids.ID]json.Uint64{}
and cast when calling math.Add64 ?

Reply:
My suggestion:
- Add a camino.go in json package
- Add a function which does the math.Add64 on json.Uint64 Types and returns json.Uint64
-> No map copying required anymore

Reply:
done, thank you for your feedback. I have already removed the loops filling the maps after the utxo loop. Instead, I am updating the balances in the main utxo loop